src/test/unit_tests/jest_tests/test_nc_upgrade_manager.test.js

```diff
@@ -673,6 +726,101 @@ describe('nc upgrade manager - upgrade config directory', () => {
             assert_upgrade_finish_or_fail_data(system_json_data, expected_data);
         });
     });
+
+    describe('nc upgrade manager - upgrade_config_dir', () => {
+        beforeEach(async () => {
+            await create_config_dir(config.NSFS_NC_DEFAULT_CONF_DIR);
+            await create_config_dir(config_root);
+            await create_redirect_file(config_fs, config_root);
+        });
+
+        afterEach(async () => {
+            await clean_config_dir(config_fs, config_root);
+        }, TEST_TIMEOUT);
+
+        it('upgrade_config_dir - empty system_data', async () => {
```

Comment:
Could you add in the title what to expect (in general in all the new titles here), for example:
```suggestion
        it('upgrade_config_dir - empty system_data - should fail', async () => {
```